sdk/storage/Azure.Storage.Blobs/src/BlobExtensions.cs

```diff
@@ -142,5 +144,39 @@ internal static string ToTagsString(this Tags tags)
                 ExpiresOn = properties.ExpiresOn,
                 IsSealed = properties.IsSealed,
             };
+
+        /// <summary>
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID.
+        /// </summary>
+        /// <returns></returns>
+        internal static IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds(IDictionary<string, string> OrIds)
+        {
+            if (OrIds == null)
+            {
+                return null;
+            }
+            if (OrIds.Count == 0 ||
+                (OrIds.Count > 0 &&
+                (OrIds.First().Key == "policy-id")))
```

Comment:
that's quite surprising condition. could you please add comment why are we doing that ?

Reply:
If we get back the object replication details of a destination blob, it will populate the dictionary with one value, where the header is "x-ms-or-policy-id" and the value is the policy id. The policy id also gets parsed and stored in the ObjectReplicationDestinationPolicy, so we don't need to continue with the parsing.

I will add a comment to detail this, cause I agree that this is not a straightforward why we are doing this.